pydantic/_internal/_generate_schema.py

```diff
@@ -1287,13 +1291,20 @@ def set_discriminator(schema: CoreSchema) -> CoreSchema:
             schema = self._apply_discriminator_to_union(schema, field_info.discriminator)
             return schema
 
+        # Convert `@field_validator` decorators to `Before/After/Plain/WrapValidator` instances:
+        validators_from_decorators = []
+        for decorator in filter_field_decorator_info_by_field(decorators.field_validators.values(), name):
+            validators_from_decorators.append(_mode_to_validator[decorator.info.mode]._from_decorator(decorator))
+
         with self.field_name_stack.push(name):
             if field_info.discriminator is not None:
-                schema = self._apply_annotations(source_type, annotations, transform_inner_schema=set_discriminator)
+                schema = self._apply_annotations(
+                    source_type, annotations + validators_from_decorators, transform_inner_schema=set_discriminator
+                )
             else:
                 schema = self._apply_annotations(
                     source_type,
-                    annotations,
+                    annotations + validators_from_decorators,
                 )
```

Comment:
Hmm, if I recall, `_apply_annotations` is one of the functions we need to take a closer look at re performance. Looks like this is adding an additional burden to that function, which I'd prefer not to do.

That being said, I don't see any performance effects in `codspeed`, so maybe it's totally fine...

Reply:
> That being said, I don't see any performance effects in `codspeed`, so maybe it's totally fine...

Would be great to review those benchmarks as it seems to not be affected by previous changes as well.

I don't think the performance hit will be noticeable, 99% of cases this only adds one annotation

Reply:
Yeah, seems negligible, I'll mark this as resolved.